CyberArk/cyberark_modules/connector_audit_logs.py

```diff
@@ -0,0 +1,208 @@
+from typing import Generator
+
+import orjson
+import requests
+from cachetools import Cache, LRUCache
+from pydantic.v1 import Field
+from sekoia_automation.checkpoint import CheckpointTimestamp, TimeUnit
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+
+from . import CyberArkModule
+from .client import ApiClient
+from .client.auth import CyberArkApiAuthentication
+from .logging import get_logger
+from .metrics import EVENTS_LAG, FORWARD_EVENTS_DURATION, INCOMING_MESSAGES, OUTCOMING_EVENTS
+
+logger = get_logger()
+
+
+class CyberArkAuditLogsConnectorConfiguration(DefaultConnectorConfiguration):
+    api_base_url: str = Field(..., description="API base url")
+    api_key: str = Field(..., description="API key", secret=True)
+    frequency: int = Field(60, description="Batch frequency in seconds")
+
+
+class CyberArkAuditLogsConnector(Connector):
+    module: CyberArkModule
+    configuration: CyberArkAuditLogsConnectorConfiguration
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.cursor = CheckpointTimestamp(
+            time_unit=TimeUnit.MILLISECOND,
+            path=self._data_path,
+            start_at=timedelta(hours=1),
+            ignore_older_than=timedelta(days=7),
+        )
+
+        self.from_date: int = self.cursor.offset
+        self.events_cache: Cache = self.load_events_cache()
+
+    def load_events_cache(self) -> Cache:
+        result: LRUCache = LRUCache(maxsize=1000)
+
+        with self.cursor._context as cache:
+            events_ids = cache.get("events_cache", [])
+
+        for event_id in events_ids:
+            result[event_id] = 1
+
+        return result
+
+    def save_events_cache(self, events_cache: Cache) -> None:
+        with self.cursor._context as cache:
+            cache["events_cache"] = list(events_cache.keys())
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            auth=CyberArkApiAuthentication(
+                auth_base_url=self.module.configuration.auth_base_url,
+                application_id=self.module.configuration.application_id,
+                client_id=self.module.configuration.login_name,
+                client_secret=self.module.configuration.password,
+                api_key=self.configuration.api_key,
+            ),
+        )
+
+    def _handle_response_error(self, response: requests.Response) -> None:
+        if not response.ok:
+            level = "critical" if response.status_code in [401, 403] else "error"
+
+            message = f"Request to CyberArk API failed with status {response.status_code} - {response.reason}"
+
+            try:
+                error = response.json()
+                logger.error(
+                    message,
+                    error=error.get("error") or error.get("message"),
+                    description=error.get("error_description") or error.get("description"),
```

Comment:
According to the specification, not OK response with only hold `message` and `description` fields.
```suggestion
                    error=error.get("message"),
                    description=error.get("description"),
```

The fields `error` and `error_description` are returned by the Authentication endpoint only.